src/libs/antares/study/binding_constraint/BindingConstraintsList.cpp

```diff
@@ -0,0 +1,383 @@
+                                            const AnyString &folder) {
+    // Log entries
+    logs.info(); // space for beauty
+    logs.info() << "Loading constraints...";
+
+    // Cleaning
+    clear();
+
+    if (study.usedByTheSolver) {
+        if (options.ignoreConstraints) {
+            logs.info() << "  The constraints have been disabled by the user";
+            return true;
+        }
+        if (!study.parameters.include.constraints) {
+            logs.info() << "  The constraints shall be ignored due to the optimization preferences";
+            return true;
+        }
+    }
+
+    EnvForLoading env(study.areas, study.header.version);
+    env.folder = folder;
+
+    env.iniFilename << env.folder << Yuni::IO::Separator << "bindingconstraints.ini";
+    IniFile ini;
+    if (!ini.open(env.iniFilename)) {
+        return false;
+    }
+
+    // For each section
+    if (ini.firstSection) {
+        for (env.section = ini.firstSection; env.section; env.section = env.section->next) {
+            if (env.section->firstProperty) {
+               auto new_bc = LoadBindingConstraint(env);
+                std::copy(new_bc.begin(), new_bc.end(), std::back_inserter(pList));
+            }
+        }
+    }
+
+    bool hasError = checkTimeSeriesWidthConsistency();
+
+    // Logs
+    if (pList.empty())
+        logs.info() << "No binding constraint found";
+    else
+    {
+        std::sort(pList.begin(), pList.end(), compareConstraints);
+
+        if (pList.size() == 1)
+            logs.info() << "1 binding constraint found";
+        else
+            logs.info() << pList.size() << " binding constraints found";
+    }
+
+    // When ran from the solver and if the simplex is in `weekly` mode,
+    // all weekly constraints will become daily ones.
+    if (study.usedByTheSolver && sorDay == study.parameters.simplexOptimizationRange) {
+        mutateWeeklyConstraintsIntoDailyOnes();
+    }
+
+    initializeTsNumbers();
+
+    return !hasError;
+}
+
+bool BindingConstraintsList::checkTimeSeriesWidthConsistency() const {
+    bool hasError = false;
+    std::map<std::string, unsigned, std::less<>> timeSeriesCountByGroup;
+    for(const auto& bc: this->pList) {
+        auto count = timeSeriesCountByGroup[bc->group()];
+        auto width = bc->RHSTimeSeries().width;
+        if (count == 0) {
+            timeSeriesCountByGroup[bc->group()] = width;
+            continue;
+        }
+        if (count != width) {
+            logs.error() << "Inconsistent time series width for constraint of the same group. Group at fault: "
+                         << bc->group()
+                         << " .Previous width was " << count
+                         << " new constraint " << bc->name()
+                         << " found with width of " << width;
+            hasError = true;
+        }
+    }
+    return hasError;
+}
+
+void BindingConstraintsList::initializeTsNumbers() {
```

Comment:
**initializeTsNumbers()** is **private** and thus cannot be called from end-to-end tests.
Should we make it **public** ? Or should it be called in another **public** function callable from tests ?

Reply:
disappeared in current **develop**